Add SSL/TLS articles to conceptual docs (#32194)
dotnet/docs: docs/core/extensions/sslstream-migration-from-framework.md

---
title: Migrating from .NET Framework to .NET
description: How to migrate code using SslStream from .NET Framework to .NET
author: rzikm
ms.author: rzikm
ms.date: 11/04/2022
---

# Migrating from .NET Framework to .NET

> .NET brought many improvements as well as breaking changes to how <xref:System.Net.Security.SslStream> works. The most important change related to <xref:System.Net.Security.SslStream> behavior is that <xref:System.Net.Security.SslStream> is no longer affected by the <xref:System.Net.ServicePointManager> class. Since .NET, allowed SSL/TLS protocols and certificate validation callbacks must be configured separately for each <xref:System.Net.Security.SslStream> instance.

Reviewer: We still honor some parts for ServicePointManager for HttpWebRequest but not for HttpClient. 
We may add more info what is supported and what not as this always causes confusion.

---

> title: Migrating from .NET Framework to .NET

Reviewer 2: ```suggestion
title: Migrate from .NET Framework to .NET
```

> description: How to migrate code using SslStream from .NET Framework to .NET

Reviewer 2: ```suggestion
description: Learn how to migrate code using SslStream in .NET Framework to .NET.
```

author: rzikm
ms.author: radekzikmund

> ms.date: 11/04/2022

Reviewer 2: ```suggestion
ms.date: 1/9/2023
```

---

> # Migrating from .NET Framework to .NET

Reviewer 2: ```suggestion
# Migrate from .NET Framework to .NET
```

> .NET Core brought many improvements as well as breaking changes to how <xref:System.Net.Security.SslStream> works. The most important change related to network security is that
> the <xref:System.Net.ServicePointManager> class has been mostly obsoleted and affects only the legacy <xref:System.Net.WebRequest> interface.

Reviewer 2: ```suggestion
.NET Core brought many improvements as well as breaking changes to how <xref:System.Net.Security.SslStream> works. The most important change related to network security is that the <xref:System.Net.ServicePointManager> class has been mostly obsoleted and affects only the legacy <xref:System.Net.WebRequest> interface.
```

> Since .NET, allowed SSL/TLS protocols and certificate validation callbacks must be configured separately for each <xref:System.Net.Security.SslStream> instance via the <xref:System.Net.Security.SslServerAuthenticationOptions> or <xref:System.Net.Security.SslClientAuthenticationOptions>. In order to configure network security options used in HTTPS in <xref:System.Net.Http.HttpClient>, you need to configure the security options in the underlying handler, such as <xref:System.Net.Http.SocketsHttpHandler.SslOptions?displayName=nameWithType>.

Reviewer 2: ```suggestion
Since .NET, allowed SSL/TLS protocols and certificate validation callbacks must be configured separately for each <xref:System.Net.Security.SslStream> instance via the <xref:System.Net.Security.SslServerAuthenticationOptions> or <xref:System.Net.Security.SslClientAuthenticationOptions>. In order to configure network security options used in HTTPS in <xref:System.Net.Http.HttpClient>, you need to configure the security options in the underlying handler, such as <xref:System.Net.Http.SocketsHttpHandler.SslOptions?displayName=nameWithType>.
```

I feel like this article is missing content, no? It's rather short. Is it the entire story for how to migrate from .NET Framework to .NET as it pertains to `SslStream`?

Author: I tried to think about more things which need to be explicitly migrated regarding SslStream but I did not remember any, @wfurt any ideas what more we could add?

Reviewer 2: Perhaps this is something that doesn't belong in its own article then? Maybe just a subsection in an existing article?

Reviewer: The migration from network is ongoing pain. I feel making it separate article helps users to find it. (even if it is short) We can possibly improve it over time but this looks like good starter to me.
